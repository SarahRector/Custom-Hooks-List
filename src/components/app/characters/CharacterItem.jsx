import React from 'react';
import PropTypes from 'prop-types';

const CharacterItem = ({ name, image_url, quote }) => (
  <div data-testid="details">
    <h1> {name} </h1>
    <img src={image_url} alt={name} />
    <p> {quote} </p>
  </div>
);

CharacterItem.propTypes = {
  name: PropTypes.string.isRequired,
  image_url: PropTypes.string,
  quote: PropTypes.string
};

export default CharacterItem;
