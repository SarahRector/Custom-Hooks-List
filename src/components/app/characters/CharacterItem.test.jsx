import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterItem from './CharacterItem';

describe('CharacterItem component', () => {
  afterEach(() => cleanup());
  it('renders CharacterItem', () => {
    const { asFragment } = render(<CharacterItem
      name="Trixie"
      image_url="test.png"
      quote="words and things" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
