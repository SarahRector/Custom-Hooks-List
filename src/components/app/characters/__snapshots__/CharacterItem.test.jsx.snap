// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem component renders CharacterItem 1`] = `
<DocumentFragment>
  <div
    data-testid="details"
  >
    <h1>
       Trixie 
    </h1>
    <img
      alt="Trixie"
      src="test.png"
    />
    <p>
       words and things 
    </p>
  </div>
</DocumentFragment>
`;
